python/ray/serve/_private/common.py

```diff
@@ -141,6 +199,251 @@ def update(
             message=message,
         )
 
+    def update_message(self, message: str):
+        return self._update(message=message)
+
+    def handle_transition(
+        self,
+        trigger: DeploymentStatusInternalTrigger,
+        message: str = "",
+        should_autoscale: bool = False,
+    ):
```

Comment:
Add a return type `DeploymentStatusInfo`

Reply:
Also, would be nice to add some docstring saying this function will create new instance of `DeploymentStatusInfo` or possibly returning the existing one. And be explicit that this will not alter the current `DeploymentStatusInfo`.